classes/core/APIResponse.php

```diff
@@ -41,7 +41,7 @@ public function withCSV(array $rows, array $columns, ?int $maxRows): self
         $this->getBody()->write($csvData);
         $this->withStatus(200);
         if (isset($maxRows)) {
-            $this->withHeader('X-Total-Count', $maxRows);
+            return $this->withHeader('X-Total-Count', $maxRows)->withHeader('Content-Type', self::RESPONSE_CSV);
```

Comment:
Let's make `$maxRows` required and always return `X-Total-Count`. That will make the API responses more predictable and simplify our code.